pil_analyzer/src/pil_analyzer.rs

```diff
@@ -184,43 +185,15 @@ impl<T: FieldElement> PILContext<T> {
                     self.to_source_ref(start),
                     name.name,
                     name.array_size,
-                    PolynomialType::Committed,
+                    SymbolKind::Poly(PolynomialType::Committed),
                     Some(definition),
                 );
             }
-            PilStatement::ConstantDefinition(_, name, value) => {
-                self.handle_constant_definition(name, value)
-            }
-            PilStatement::LetStatement(start, name, None) => {
-                // Handle all let statements without assignment as witness column declarations for now.
-                self.handle_polynomial_definition(
-                    self.to_source_ref(start),
-                    name,
-                    None,
-                    PolynomialType::Committed,
-                    None,
-                );
+            PilStatement::ConstantDefinition(start, name, value) => {
+                self.handle_generic_definition(start, name, Some(value))
```

Comment:
I think I'd prefer keeping `handle_constant_definition` and call it from `handle_generic_definition`, or is there a reason not to do that?

Looking at `handle_generic_definition`:
- Since `value` is `Some`, the first branch doesn't apply
- Then it checks for a lambda expression, but this probably can't happen because it wouldn't have been parsed?
- Then, if the RHS doesn't evaluate to a constant, instead of failing, we'd add a definition of a symbol of type "other". Is that what we want?

Reply:
Yeah, I can restrict it again.